### PR TITLE
Replace GameObject.SendMessage("Method") with event delegates

### PR DESCRIPTION
**Pull request by HodgsonSDAS**
Pull request for issue https://github.com/Microsoft/HoloToolkit-Unity/issues/224

Many experienced Unity developers would probably tell you not use [GameObject.SendMessage("String");](https://docs.unity3d.com/ScriptReference/GameObject.SendMessage.html) because [it's slow](http://answers.unity3d.com/questions/327888/is-sendmessage-slow-.html) (under the hood it uses reflection), hard to debug, and it lacks direct symbol references in the editor which makes it prone to spelling errors _because_ it's a string and not a symbol. This can also make it difficult to find everywhere the method referenced can be called and could potentially increase the amount of spaghetti code.  It may be worth while to see this updated to utilize a [delegate/Event call instead](https://unity3d.com/learn/tutorials/topics/scripting/events?playlist=17117).

## TL;DR
This will make the code easier to follow in the editor, easier to debug, and will increase performance.

## [Some arguments for using a delegate vs SendMessage](http://answers.unity3d.com/questions/185980/sendbroadcast-message-vs-delegates-and-events.html):

> SendMessage and delegates are used for similar purposes, but they work entirely differently. Delegates/Events behave exactly like they do in any other .Net program. SendMessage() is called much like Unity calls other messages like Update(). Unity reflects over all the code on the game object to see if it can invoke any messages.
> 
> As far as performance. Delegates are several times faster than SendMessage. In the order of performance:
> 
`(Least) Direct Method Call ==> Delegate/Event ==> Update() ==> SendMessage (Most Expensive)`
> As far as usage, SendMessage is like a very broad direct method call. The general paradigm is that one script specifically wants to send a message to a specific object. We're just not sure how many times is can respond or if it can respond at all. We know the intended receiver, we just don't know if it can receive the message.
> 
> With events, I usually follow the idea that the event owner doesn't know who or how many objects might respond. We are simply telling the world that something happened without too much knowledge of who will use this information.

## Changes to the following classes:

GestureManager.cs

- [x] Change System.Actions to delegate events to fix delegate subtraction issues in GestureManipulator.cs
- [x] Add OnSelect(GameObject) event
- [x] Add explicit private modifiers to Unity Methods

GestureManipulator.cs

- [x] Remove redundant using HoloToolkit.Unity reference
- [x] Ensure we're checking GesureManager instance is null
- [x] Add explicit private modifiers to Unity Methods

GazeManager.cs

- [x] Remote unused using reference
- [x] Add delegate events for OnGazeEnter
- [x] Add delegate events for OnGazeExit
- [x] Ensure StabilizationPlaneModifier instance is null

TapToPlace.cs

- [x] Update to work with new design pattern

GestureResponder.cs

- [x] Update to work with new design pattern

SelectedObjectMessageSender.cs

- [x] Remove File

SelectedObjectMessageReciever.cs

- [x] Rename file to ChangeColorOnFocus
- [x] Update to work with new design pattern

FocusedObjectMessageSender.cs

- [x] Remove File

FocusedObjectMessageReciever.cs

- [x] Rename file
- [x] Update to work with new design pattern

OnSelectEvent.cs

- [x] Remove File (replaced by Intractable.cs)

OnGazeEnterEvent.cs

- [x] Remove File (replaced by Intractable.cs)

OnGazeLeaveEvent.cs

- [x] Remove File (replaced by Intractable.cs)

Intractable.cs

- [x] Add file Intractable.cs

KeywordManager.cs

- [x] Update to work with new design pattern

ScaleObjectMessageReciever.cs

- [x] Update to work with new design pattern
- [x] Renamed to ScaleFocusedObject.cs

SphereKeywords.cs

- [x] Update to work with new design pattern

## Updates to the following scenes:

SelectedObjectKeywords.unity
FocustedObjectKeywords.unity
GazeEvents.unity

## Updates to the following Prefabs:

SelectedObjectKeywordManager.prefab
FocusedObjectKeywordManager.prefab

### CODE REVIEW

**Comment by HodgsonSDAS** [comment]
messed up my branch. I'll close and reopen.
